### ISSUE TITLE
Nightcode (or, possibly, JVM) crashes

### ISSUE THREAD

**Issue by phunni** [oakes/Nightcode#256]
While Nightcode was running, but I was away from the keyboard (plugging in my phone  - but I doubt that's a factor) Nightcode crashed withthe following error messages:

```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f3265500725, pid=22139, tid=0x00007f333428b700
#
# JRE version: OpenJDK Runtime Environment (8.0_212-b01) (build 1.8.0_212-b01)
# Java VM: OpenJDK 64-Bit Server VM (25.212-b01 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libjfxwebkit.so+0x181d725]
#
# Core dump written. Default location: /home/paul/core or core.22139
#
# An error report file with more information is saved as:
# /home/paul/hs_err_pid22139.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
/usr/bin/nightcode: line 2: 22139 Aborted                 (core dumped) java -Xmx512m -jar /usr/share/java/nightcode/nightcode-standalone.jar $@
```
Obviously, I'm not certain if this is a problem with the JVM or with Nightcode or even my system, but I thought it might be helpful to report.
I'm running Nightcode on Arch Linux, 64bit on Openbox.

**Comment by oakes** [comment]
Are you using the Linux installer or the JAR file? I would suggest updating your JDK to 11 or above and using the latest JAR file [on the website](https://sekao.net/nightcode/). The webview in Java 8 is quite old and buggy.

**Comment by beenotung** [comment]
I am facing the same/similar issue.
It was working fine on my laptop but encountered this problem on desktop (both are running ArchLinux)

I have tried with jdk 8 and jdk 11, also tried to install from aur and using the Nightcode-2.7.0.jar linked from the official website (download link was pointing to github).

It was working fine when I was using it to open clj files, but when I used it to open readme, it crashed. If I restart the editor, it will auto open that readme file so will crash again. A workaround is to delete the readme file before opening the IDE.

Attached the log file.

[hs_err_pid211090.log](https://github.com/oakes/Nightcode/files/3638566/hs_err_pid211090.log)

**Comment by oakes** [comment]
@beenotung can you try this jar out and let me know if it fixes the issue? https://github.com/oakes/Nightcode/releases/download/2.7.1-SNAPSHOT/nightcode-2.7.1-SNAPSHOT-standalone.jar

It has the most recent version of java fx so the webview might be better.

**Comment by oakes** [comment]
I ended up cutting a new release so please try this instead:

https://github.com/oakes/Nightcode/releases/download/2.7.1/Nightcode-2.7.1.jar

**Comment by oakes** [comment]
Correction....i just released 2.8.0, use this link instead: https://github.com/oakes/Nightcode/releases/download/2.8.0/Nightcode-2.8.0.jar

**Comment by oakes** [comment]
There is now a native .AppImage package for linux that includes JDK 11, so i'll close this. The crash was most likely due to the bugginess of the JDK 8 webview.
